### PR TITLE
Run-time config knobs

### LINKED ISSUE #15850: Document config knobs for .NET Core
Document supported run-time configuration knobs for .NET Core apps. Include runtimeconfig.json and environment variables for each, as well as the corresponding .NET Framework setting (app.config). Also include the first version it was available in.

Reference:

- https://github.com/dotnet/coreclr/blob/ed5dc831b09a0bfed76ddad684008bebc86ab2f0/src/vm/eeconfig.cpp
- https://github.com/dotnet/coreclr/pull/25657
- https://github.com/dotnet/coreclr/blob/master/src/inc/clrconfigvalues.h
- https://devblogs.microsoft.com/dotnet/the-history-of-the-gc-configs/

Related to #13425.

### PR DESCRIPTION
**Pull request by gewarren**
Add configuration knobs for .NET Core through runtimeconfig.json or environment variables.

Fixes #15850 

Preview links:

[Compilation](https://review.docs.microsoft.com/en-us/dotnet/core/run-time-config/compilation)
[Debugging and profiling](https://review.docs.microsoft.com/en-us/dotnet/core/run-time-config/debugging-profiling?branch=pr-en-us-16019)
[Globalization](https://review.docs.microsoft.com/en-us/dotnet/core/run-time-config/globalization?branch=pr-en-us-16019)
[Networking](https://review.docs.microsoft.com/en-us/dotnet/core/run-time-config/networking?branch=pr-en-us-16019)
[Threading](https://review.docs.microsoft.com/en-us/dotnet/core/run-time-config/threading?branch=pr-en-us-16019)

### CODE REVIEW

**Comment by gewarren** [comment]
@jkotas Could you review please? Some of this you previously reviewed in #12959, but I pulled it out of that PR before it was merged.

File: docs/core/run-time-config/debugging-profiling.md
Comment (gewarren) [inline docs/core/run-time-config/debugging-profiling.md]: @jkotas Should this have the COMPlus_ prefix?

File: docs/core/run-time-config/debugging-profiling.md
Comment (gewarren) [inline docs/core/run-time-config/debugging-profiling.md]: @jkotas Should this have the COMPlus_ prefix?

File: docs/core/run-time-config/debugging-profiling.md
Comment (gewarren) [inline docs/core/run-time-config/debugging-profiling.md]: @jkotas Should these have the COMPlus_ prefix?

File: docs/core/run-time-config/globalization.md
Comment (Thraka) [inline docs/core/run-time-config/globalization.md]: Capitalization of `gannen` is different in a few places. Is it `Gannen` or `gannen`?

File: docs/core/run-time-config/compilation.md
Comment (Thraka) [inline docs/core/run-time-config/compilation.md]: Are these settings for the .net core **runtime** or settings for the app at **run-time**?

File: docs/core/run-time-config/threading.md
Comment (Thraka) [inline docs/core/run-time-config/threading.md]: Should these empty entries have some sort of text that states it's not configurable? Some of the ones in the other files have `N/A`

File: docs/core/run-time-config/globalization.md
Comment (Thraka) [inline docs/core/run-time-config/globalization.md]: Should the 0 and 1 in all of these (even where it lists the value: Default (0) ) be in enclosed in code elements? Since it's the value to enter, I think it should be on par with false or true.

**Review by Thraka — APPROVED** [review]
LGTM. Some things to consider for consistency.

File: docs/core/run-time-config/debugging-profiling.md
Comment (jkotas) [reply docs/core/run-time-config/debugging-profiling.md]: No, all variables related to profiling are special - no `COMPlus_` prefix.

File: docs/core/run-time-config/networking.md
Comment (jkotas) [inline docs/core/run-time-config/networking.md]: We are removing these settings for .NET 5. Can we update the docs for this?

cc @ManickaP

**Comment by jkotas** [comment]
LGTM modulo comments.

File: docs/core/run-time-config/compilation.md
Comment (gewarren) [reply docs/core/run-time-config/compilation.md]: Good question. I'd assumed they were settings for the app at run time, but now that I think about it, they would affect any .NET Core app that's running at the time on the computer. So maybe they are settings for the .NET Core runtime.

File: docs/core/run-time-config/networking.md
Comment (gewarren) [reply docs/core/run-time-config/networking.md]: We aren't supposed to talk about future functionality in docs, so I don't think we need to mention it yet.

File: docs/core/run-time-config/compilation.md
Comment (gewarren) [reply docs/core/run-time-config/compilation.md]: Thinking about it more, it seems like the environment variables would affect any running .NET Core app, but the runtimeconfig.json file is app-specific. So I believe these are settings for the app at run time.

File: docs/core/run-time-config/networking.md
Comment (jkotas) [reply docs/core/run-time-config/networking.md]: What is the right way to track updating the docs with this info once .NET 5 ships?

Comment (gewarren) [reply docs/core/run-time-config/networking.md]: I created an issue [here](https://github.com/dotnet/docs/issues/16088) and labeled it with a new label, Release: .NET 5. @mairaw Please advise if I should do something different instead.

**Comment by gewarren** [comment]
Thanks for the reviews @jkotas and @Thraka!

**Pull request merged**
